spec/lib/ansible/runner_spec.rb

```diff
@@ -57,20 +88,33 @@
 
   describe ".run_async" do
     let(:playbook) { "/path/to/my/playbook" }
-    before { expect(File).to receive(:exist?).with(playbook).and_return(true) }
+    before do
+      allow(File).to receive(:exist?).and_call_original
+      allow(File).to receive(:exist?).with(playbook).and_return(true)
+    end
 
     it "calls ansible-runner with start" do
-      expected_extra_vars = "--extra-vars\\ \\\\\\{\\\\\\\"id\\\\\\\":\\\\\\\"201ac780-7bf4-0136-3b9e-54e1ad8b3cf4\\\\\\\"\\\\\\}"
-      expected_command_line = [
-        "ansible-runner start",
-        "--json --playbook #{playbook} --ident result --hosts localhost --cmdline #{expected_extra_vars}"
-      ]
+      expect(AwesomeSpawn).to receive(:run) do |command, options|
+        expect(command).to eq("ansible-runner")
+        expect(options[:env]).to eq(env_vars)
+
+        method, dir, json, args = options[:params]
 
-      expect(AwesomeSpawn).to receive(:launch)
-        .with(env_vars, a_string_including(*expected_command_line), {})
+        expect(method).to eq("start")
+        expect(json).to   eq(:json)
+        expect(args).to   eq(:ident => "result", :playbook => playbook)
 
-      result = described_class.run_async(env_vars, extra_vars, playbook)
-      expect(result).kind_of?(Ansible::Runner::ResponseAsync)
+        hosts = File.read(File.join(dir, "inventory", "hosts"))
+        expect(hosts).to eq("localhost")
+
+        extravars = JSON.parse(File.read(File.join(dir, "env", "extravars")))
+        expect(extravars).to eq("id" => uuid, "ansible_connection" => "local")
+
+        expect(File.exist?(File.join(dir, "env", "cmdline"))).to be_falsey
```

Comment:
Super minor, but readability wise, it would be nice if dir were a Pathname, then lines like this become...

```ruby
dir.join("env", "cmdline").exist?
```

Reply:
So this is actually the parameter being passed through to `AwesomeSpawn`. So I would have to either create the pathname object in a separate step from the parameter or change the `Runner` module to pass a pathname.

I think I'd rather keep it as is (FWIW, if `dir` was a local `let` variable or something then I would definitely agree).